x/ecocredit/simulation/marketplace/operation.go

```diff
@@ -0,0 +1,409 @@
+		if spendable == nil {
+			return op, nil, err
+		}
+
+		txCtx := simulation.OperationInput{
+			R:               r,
+			App:             app,
+			TxGen:           simapp.MakeTestEncodingConfig().TxConfig,
+			Cdc:             nil,
+			Msg:             msg,
+			MsgType:         msg.Type(),
+			Context:         sdkCtx,
+			SimAccount:      *account,
+			AccountKeeper:   ak,
+			Bankkeeper:      bk,
+			ModuleName:      ecocredit.ModuleName,
+			CoinsSpentInMsg: spendable,
+		}
+
+		return utils.GenAndDeliverTxWithRandFees(txCtx)
+
+	}
+}
+
+// SimulateMsgSell generates a Marketplace/MsgSell with random values.
+func SimulateMsgSell(ak ecocredit.AccountKeeper, bk ecocredit.BankKeeper,
+	qryClient core.QueryClient) simtypes.Operation {
+	return func(
+		r *rand.Rand, baseApp *baseapp.BaseApp, sdkCtx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		owner, _ := simtypes.RandomAcc(r, accs)
+		ownerAddr := owner.Address.String()
+
+		ctx := regentypes.Context{Context: sdkCtx}
+		class, op, err := utils.GetRandomClass(sdkCtx, r, qryClient, TypeMsgSell)
+		if class == nil {
+			return op, nil, err
+		}
+
+		batchRes, err := qryClient.BatchesByIssuer(ctx, &core.QueryBatchesByIssuerRequest{Issuer: owner.Address.String()})
+		if err != nil {
+			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSell, err.Error()), nil, err
+		}
+
+		batches := batchRes.Batches
+		if len(batches) == 0 {
+			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSell, "no credit batches"), nil, nil
+		}
+		max := 1
+		if len(batches) > 1 {
+			max = simtypes.RandIntBetween(r, 1, len(batches))
+		}
+
+		sellOrders := make([]*marketplace.MsgSell_Order, max)
+		for i := 0; i < max; i++ {
+			bal, err := qryClient.Balance(ctx, &core.QueryBalanceRequest{Account: ownerAddr, BatchDenom: batches[i].BatchDenom})
+			if err != nil {
+				return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSell, err.Error()), nil, err
+			}
+
+			exp := ctx.BlockTime().AddDate(1, 0, 0)
+			d, err := math.NewNonNegativeDecFromString(bal.Balance.Tradable)
+			if err != nil {
+				return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSell, err.Error()), nil, err
+			}
+
+			if d.IsZero() {
+				return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSell, "no balance"), nil, nil
+			}
+
+			balInt, err := d.Int64()
+			if err != nil {
+				return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSell, err.Error()), nil, err
+			}
+
+			quantity := int(balInt)
+			if balInt > 1 {
+				quantity = simtypes.RandIntBetween(r, 1, int(quantity))
+			}
+
+			askPrice := sdk.NewInt64Coin(sdk.DefaultBondDenom, int64(simtypes.RandIntBetween(r, 1, 50)))
+			sellOrders[i] = &marketplace.MsgSell_Order{
+				BatchDenom:        batches[i].BatchDenom,
+				Quantity:          fmt.Sprintf("%d", quantity),
+				AskPrice:          &askPrice,
+				DisableAutoRetire: r.Int63n(101) <= 30,
```

Comment:
Might be nice to add a comment as you did below such as `// 30% chance of disable auto-retire`